examples/ecs/custom_query_param.rs

```diff
@@ -0,0 +1,189 @@
+use bevy::{
+    ecs::{
+        component::Component,
+        query::{Fetch, FilterFetch},
+    },
+    prelude::*,
+};
+use std::{fmt::Debug, marker::PhantomData};
+
+/// This examples illustrates the usage of `Fetch` and `FilterFetch` derive macros, that allow
+/// defining custom query and filter types.
```

Comment:
Could you include `Entity` in one of the derived queries? It's a very useful bit of functionality and isn't always obvious that it works the same way as querying for components.

Reply:
It's already included in the first query example.

Reply:
Ah, my bad!